src/importexport/tabledit/internal/importtef.cpp

```diff
@@ -0,0 +1,1045 @@
+    }
+    if (!tefHeader.copyright.empty()) {
+        score->setMetaTag(u"copyright", String::fromUtf8(tefHeader.copyright.c_str()));
+    }
+}
+
+// convert the reading list into repeat and (todo) codas, segnos and voltas
+
+void TablEdit::createRepeats()
+{
+    LOGD("reading list size %zu number of measures %zu", tefReadingList.size(), tefMeasures.size());
+    // proof of concept: add repeat to whole score if
+    // - reading list contains two items
+    // - both spanning the entire score
+    if (tefReadingList.size() == 2
+        && tefReadingList.at(0).firstMeasure == 1 && tefReadingList.at(0).lastMeasure == static_cast<int>(tefMeasures.size())
+        && tefReadingList.at(1).firstMeasure == 1 && tefReadingList.at(1).lastMeasure == static_cast<int>(tefMeasures.size())
+        ) {
+        LOGD("do it");
+        if (score->measures()->empty()) {
+            LOGE("no measures in score");
+            return;
+        }
+        Measure* first { score->firstMeasure() };
+        Measure* last { score->lastMeasure() };
+        first->setRepeatStart(true);
+        last->setRepeatEnd(true);
+    }
+    else {
+        LOGD("no score repeat");
+    }
+}
+
+void TablEdit::createScore()
+{
+    createProperties();
+    createParts();
+    createTitleFrame();
+    createMeasures();
+    createNotesFrame();
+    createContents();
+    createRepeats();
+    createTexts();
+    createLinkedTabs();
+}
+
+void TablEdit::createTempo()
+{
+    mu::engraving::Measure* measure = score->firstMeasure();
+    mu::engraving::Segment* segment = measure->getSegment(mu::engraving::SegmentType::ChordRest, mu::engraving::Fraction(0, 1));
+    mu::engraving::TempoText* tt = new mu::engraving::TempoText(segment);
+    tt->setTempo(double(tefHeader.tempo) / 60.0);
+    tt->setTrack(0);
+    tt->setFollowText(true);
+    muse::String tempoText = mu::engraving::TempoText::duration2tempoTextString(mu::engraving::DurationType::V_QUARTER);
+    tempoText += u" = ";
+    tempoText += muse::String::number(tefHeader.tempo);
+    tt->setXmlText(tempoText);
+    segment->add(tt);
+}
+
+void TablEdit::createTexts()
+{
+    for (const auto& textMarker : tefTextMarkers) {
+        LOGD("position %d string %d text marker %d", textMarker.position, textMarker.string, textMarker.index);
+        bool ok { true };
+        const auto part = partIdx(textMarker.string, ok);
+        if (!ok) {
+            LOGD("error: invalid string %d", textMarker.string);
+            continue;
+        }
+        const auto track = part * VOICES;
+        LOGD("part %zu track %zu", part, track);
+
+        Fraction tick { textMarker.position, 64 }; // position is in 64th
+        Measure* measure { tick2measure(score, tick) };
+        if (!measure) {
+            LOGD("error: no measure");
+            continue;
+        }
+        else {
+            LOGD("measure %p", measure);
+        }
+        Segment* segment { measure->getSegment(mu::engraving::SegmentType::ChordRest, tick) };
```

Comment:
Shouldn't this be `Segment::CHORD_REST_OR_TIME_TICK_TYPE`, for the situation that the text doesn't coincide with a Chord? Or will it always do so?

Reply:
TablEdit happily generates texts at ticks without chord, which currently imports without obvious ill effects. You may very well be right, I am not sure.